third_party/amd/lib/TritonAMDGPUToLLVM/Utility.cpp

```diff
@@ -137,4 +137,63 @@ Value llGetPid(Location loc, ConversionPatternRewriter &rewriter,
   return rewriter.create<arith::IndexCastOp>(loc, i32_ty, blockId);
 }
 
+Value llLoad(ConversionPatternRewriter &rewriter, Location loc, Value ptr,
+             Type elemTy, Value pred, unsigned vecStart,
+             SmallVector<Value> otherElems) {
+  Block *currentBlock = rewriter.getInsertionBlock();
+  Block *afterLoad =
+      rewriter.splitBlock(currentBlock, rewriter.getInsertionPoint());
+  afterLoad->addArgument({elemTy}, {loc});
+  Block *trueBlock = rewriter.createBlock(afterLoad);
+  Block *falseBlock =
+      rewriter.splitBlock(trueBlock, rewriter.getInsertionPoint());
+  rewriter.setInsertionPointToEnd(currentBlock);
+  rewriter.create<LLVM::CondBrOp>(loc, pred, trueBlock, falseBlock);
+  rewriter.setInsertionPointToStart(trueBlock);
+  auto loadOp = rewriter.create<LLVM::LoadOp>(loc, elemTy, ptr);
+  rewriter.create<LLVM::BrOp>(loc, loadOp->getResult(0), afterLoad);
+  rewriter.setInsertionPointToStart(falseBlock);
+  auto valueElemTy = getElementTypeOrSelf(elemTy);
+  mlir::Attribute zero = rewriter.getZeroAttr(valueElemTy);
+  auto denseValue =
+      DenseElementsAttr::get(elemTy.cast<mlir::ShapedType>(), zero);
+  Value zeroVal = rewriter.create<LLVM::ConstantOp>(loc, elemTy, denseValue);
+  Value falseVal = zeroVal;
+  // If we need to mask the loaded value with other elements
+  if (otherElems.size() != 0) {
+    auto vecTy = dyn_cast<VectorType>(elemTy);
+    assert(vecTy && "Expected vector type");
+    assert(ptr.getType().cast<LLVMPointerType>().getAddressSpace() == 0 &&
+           "Expected to only mask global memory loads");
+    auto vec = vecTy.getNumElements();
+    Value v = undef(elemTy);
+    for (size_t s = 0; s < vec; ++s) {
+      Value otherElem = otherElems[vecStart + s];
+      Value indexVal = rewriter.create<LLVM::ConstantOp>(
```

Comment:
Couldn't use `createIndexAttrConstant` here since it's a non-static function of `ConvertOpToLLVM`, but lmk if there's a better way. Seems benign given that this is a 1 liner.